### ISSUE TITLE
Extend `vscode.extensions.getExtension` to return extensions from other extension host

### ISSUE THREAD

**Issue by jrieken** [microsoft/vscode#133734]
Today, `getExtension` only returns extensions that are registered in the same extension host. This can be problematic when extension enable/disable feature in the presence of other extensions. 

slightly related: https://github.com/microsoft/vscode/issues/84997

**Comment by aganbokeng** [comment]
vscode.extensions.getExtension

**Comment by thegecko** [comment]
Is the intent that this feature would add the ability to use extension APIs across extension hosts?

**Comment by yiliang114** [comment]
Is there any progress on this issue so far?

**Comment by jrieken** [comment]
> Is the intent that this feature would add the ability to use extension APIs across extension hosts?

No, it just expose metadata about extensions from other extension host but you will not (ever) be able to use their API (because there are guarantees that these API are IPC-safe)

**Comment by alexdima** [comment]
🤖 Thanks for filing this. The work here is fully covered by https://github.com/microsoft/vscode/issues/145307, which introduced the proposed `extensionsAny` API — including `getExtension(id, includeDifferentExtensionHosts)` and `extensions.allAcrossExtensionHosts`. Consolidating discussion there as that issue tracks finalization.

/duplicate of #145307
